packages/db/fauna/resources/Function/createPinRequest.js

```diff
@@ -0,0 +1,71 @@
+import fauna from 'faunadb'
+
+const {
+  Function,
+  CreateFunction,
+  Query,
+  Lambda,
+  Let,
+  Select,
+  Var,
+  If,
+  Create,
+  Update,
+  Exists,
+  Now,
+  Abort,
+  Not,
+  Index,
+  IsEmpty,
+  Get,
+  Match,
+  IsNull,
+  Call
+} = fauna
+
+const name = 'createPinRequest'
+const body = Query(
+  Lambda(
+    ['data'],
+    Let(
+      {
+        cid: Select('cid', Var('data')),
+        content: Call('findContentByCid', Var('cid')),
+        contentRef: Select('ref', Var('content')),
```

Comment:
Can return `null` if not found.

```suggestion
        contentRef: If(
          IsNull(Var('content')),
          Abort('content not found'),
          Select('ref', Var('content'))
        )
```

So, probably can remove the `Not(Exists(Var('contentRef')))` later.